### DISCUSSION TITLE
How to use local documents

### DISCUSSION THREAD

**Discussion by ShutupBee** [nlmatics/llmsherpa#85] [Q&A]
I would like to ask how to use local documentation? I tried to replace the link of pdf_url with the path of my local document, but the error LocationValueError: No host specified. I don't know how to modify, can you help me?

**Comment by hellerphilipp** [comment]
You can try one of two approaches: 

Either, you pass a qualified path to the local file to `read_pdf` (see [os.path documentation](https://docs.python.org/3/library/os.path.html)):
```python
doc = layout_pdf_reader_instance.read_pdf(filepath_full)
```

or, alternatively, you read the file into a byte buffer and pass that as a file:
```python
with open(filepath_full, "rb") as file_bytes:
    doc = layout_pdf_reader_instance.read_pdf("file.pdf", contents=file_bytes.getvalue())
```

( I didn't test this code specifically, but I remember having implemented it in a similar fashion )
